src/main/java/uk/ac/ebi/tsc/portal/api/team/service/TeamService.java

```diff
@@ -883,4 +884,35 @@ public Team checkIfOwnerOrManagerOfTeam(String teamName,
 		return team;
 	}
 
-}
\ No newline at end of file
+	public Team setContactEmails(Set<String> emails, Team team) {
+		Set<String> existingEmails = team.getTeamContactEmails();
+		existingEmails.addAll(emails);
+		team.setTeamContactEmails(existingEmails);
+		team = this.save(team);
+		return team;
+	}
+
+	public Team removeTeamContactEmail(Team team, String userEmail) {
+		team.getTeamContactEmails().remove(userEmail);
+		team = this.save(team);
+		return team;
+	}
+
+	public Collection<TeamResource> populateTeamContactEmails(Collection<Team> teams, Principal principal) {
+
+		Collection<TeamResource> teamResources = teams.stream().map(TeamResource::new).collect(Collectors.toList());
+		teamResources.forEach(tr -> {
+			populateTeamContactEmails(tr, principal);
+		});
+		return teamResources;
+	}
+
+	public TeamResource populateTeamContactEmails(TeamResource teamResource, Principal principal) {
+
+		Team team = findByDomainReference(teamResource.getDomainReference());
+		if (team.getAccount().getUsername().equals(principal.getName())) {
+			teamResource.setTeamContactEmails(team.getTeamContactEmails());
+		}
+		return teamResource;
+	}
+}
```

Comment:
This method in the end is called in just one place. Over there it looks like you have access to `Team` entity with -most probably - a populated Account Entity. Correct or not? If so, why do you have to loose that information that you have already and do another query to retrieve it again. What happens here is: find a team -> convert it to resource -> find a team again.

Reply:
Have refactored this.